File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/mount/MountEquipedErrorMessage.scala





















// Generated on 05/03/2013 10:34:41
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait MountEquipedErrorMessage extends MessageDeserializer {
  val messageId = 5963

  protected trait Message extends BaseMessage with Serializable {
    val messageId = MountEquipedErrorMessage.messageId

    val errorType: Byte

    def serialize(buf: Buffer) {
      buf.writeByte(errorType)
    }

  }

  protected def create(errorType: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val errorType = buf.readByte()

    create(errorType)
  }

}

object MountEquipedErrorMessage extends MountEquipedErrorMessage {
  case class MessageImpl(errorType: Byte) extends Message
  type Target = MessageImpl

  override def create(errorType: Byte) = MessageImpl(errorType)

  def apply(errorType: Byte) = create(errorType)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/MountEquipedErrorMessage.java


// Generated on 05/08/2013 19:37:44
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class MountEquipedErrorMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5963;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte errorType;
    
    public MountEquipedErrorMessage() { }
    
    public MountEquipedErrorMessage(byte errorType) {
        this.errorType = errorType;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(errorType);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        errorType = buf.readByte();
        if (errorType < 0)
            throw new RuntimeException("Forbidden value on errorType = " + errorType + ", it doesn't respect the following condition : errorType < 0");
    }
    
}
